go/chat/search/indexer.go

```diff
@@ -201,20 +330,34 @@ type reindexOpts struct {
 // GetMessages api to fill in the holes. If our index is missing many messages,
 // we page through and add batches of missing messages.
 func (idx *Indexer) reindexConv(ctx context.Context, rconv types.RemoteConversation, uid gregor1.UID,
-	convIdx *chat1.ConversationIndex, opts reindexOpts) (completedJobs int, newIdx *chat1.ConversationIndex, err error) {
+	opts reindexOpts) (completedJobs int, err error) {
 
 	conv := rconv.Conv
+	convIdx, lock, err := idx.AcquireConvIndex(ctx, conv.GetConvID(), uid)
+	if err != nil {
+		return 0, err
+	}
 	missingIDs := convIdx.MissingIDForConv(conv)
+	lock.Release(ctx)
 	if len(missingIDs) == 0 {
-		return 0, convIdx, nil
+		return 0, nil
 	}
 	minIdxID := missingIDs[0]
 	maxIdxID := missingIDs[len(missingIDs)-1]
 
 	convID := conv.GetConvID()
 	defer idx.Trace(ctx, func() error { return err },
-		fmt.Sprintf("Indexer.reindex: conv: %v, minID: %v, maxID: %v, numMissing: %v",
+		fmt.Sprintf("reindexConv: conv: %v, minID: %v, maxID: %v, numMissing: %v",
 			rconv.GetName(), minIdxID, maxIdxID, len(missingIDs)))()
+	defer func() {
```

Comment:
let's ditch this i don't think it's worth to log every time unless we're debugging